lib/src/components/NxSegmentedButton/stateful/__tests__/NxStatefulSegmentedButton.test.tsx

```diff
@@ -4,207 +4,309 @@
+      const defaultBtn = quickRender().getByRole('button', { name: 'Click Me' }),
+          customBtn = quickRender({ type: 'button'}).getByRole('button', { name: 'Click Me' }),
+          customSubmitBtn = quickRender({ type: 'submit' }).getByRole('button', { name: 'Click Me' });
+
+      expect(defaultBtn).not.toHaveAttribute('type', 'button');
+      expect(customBtn).toHaveAttribute('type', 'button');
+      expect(customSubmitBtn).toHaveAttribute('type', 'submit');
     });
-    mounted.update();
-    expect(mounted.find('.nx-dropdown-menu')).not.toExist();
-    mounted.unmount();
-  });
 
-  it('closes the dropdown when the Escape key is pressed on this component', function() {
-    const mounted = getMountedComponent(undefined, { attachTo: container });
+    it('sets the onClick handler on the main button', async function() {
+      const user = userEvent.setup(),
+          onClick = jest.fn(),
+          component = quickRender({ onClick }),
+          mainBtn = component.getByRole('button', { name: 'Click Me' }),
+          dropdownToggleBtn = component.getByRole('button', { name: 'more options' });
 
-    act(() => {
-      mounted.find('button.nx-segmented-btn__dropdown-btn').getDOMNode()
-          .dispatchEvent(new MouseEvent('click', { bubbles: true }));
-    });
-    mounted.update();
-    expect(mounted.find('.nx-dropdown-menu')).toExist();
+      expect(onClick).not.toHaveBeenCalled();
 
-    act(() => {
-      mounted.find(NxSegmentedButton).getDOMNode()
-          .dispatchEvent(new KeyboardEvent('keydown', {key: 'Escape', bubbles: true }));
-    });
-    mounted.update();
-    expect(mounted.find('.nx-dropdown-menu')).not.toExist();
-
-    mounted.unmount();
-  });
-
-  it('closes the dropdown when an outside click happens', function() {
-    let element: ReactWrapper;
-
-    // `act` ensures that updates related to “units” of interaction with a user interface have been processed
-    // and applied to the DOM before you make any assertions.
-    // See https://reactjs.org/blog/2019/02/06/react-v16.8.0.html#testing-hooks
-    // and https://reactjs.org/docs/testing-recipes.html#act
-    act(() => {
-      element = mount(
-        <div>
-          <NxStatefulSegmentedButton { ...minimalProps }/>
-          <button id="test-btn">click</button>
-        </div>,
-        { attachTo: container }
-      );
-    });
+      // confirm onClick handler is not on dropdown button
+      await act(async () => { await user.click(dropdownToggleBtn); });
+      expect(onClick).not.toHaveBeenCalled();
 
-    act(() => {
-      // Jest/JSDom need actual events to be able to trigger effects properly
-      // See https://stackoverflow.com/questions/27557624/simulating-click-on-document-reactjs-jsdom
-      element.find('button.nx-segmented-btn__dropdown-btn').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
+      await act(async () => { await user.click(mainBtn); });
```

Comment:
In some situations in the stateful conversion, I was getting the `Warning: An update to ForwardRef(nxStatefulSegmentedButton) inside a test was not wrapped in act(...)`. It usually seemed to happen when closing the dropdown, although it was also necessary in this test.